### PR TITLE
Dragen transcriptome pipeline update

### PR DESCRIPTION
**Pull request by skanwal**
- added dragen-v4-WTS tool definition
- added dragen transcriptome pipeline that uses typescript
- initialised both on ICA - development

Tested on ICA with wfr.17f8e450fdca44c1b2fc770580db7b4c.

### CODE REVIEW

File: tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+            --directory "$(get_ref_mount())" \\
+            --extract \\
+            --file "$(inputs.reference_tar.path)"
+
+          # Run dragen command and import options from cli
+          $(get_dragen_eval_line())
+
+      - |
+        ${
+            return generate_germline_mount_points(inputs);
```
Comment (alexiswl) [inline tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:77]: If you want, we can create a function called generate_transcriptome_mount_points that calls generate_germline_mount_points underneath?

File: tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+# ILMN V1 Resources Guide: https://illumina.gitbook.io/ica-v1/analysis/a-taskexecution#type-and-size
+# ILMN V2 Resources Guide: https://help.ica.illumina.com/project/p-flow/f-pipelines#compute-types
+# ILMN Resources Guide: https://illumina.gitbook.io/ica-v1/analysis/a-taskexecution#type-and-size
+hints:
+    ResourceRequirement:
+        ilmn-tes:resources:
+            tier: standard
+            type: fpga
+            size: medium
```
Comment (alexiswl) [inline tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:31]: From 
```yaml
hints:
    ResourceRequirement:
        ilmn-tes:resources:
            tier: standard
            type: fpga
            size: medium
```

To 

```yaml
hints:
    ResourceRequirement:
        ilmn-tes:resources:tier: standard
        ilmn-tes:resources:type: fpga
        ilmn-tes:resources:size: medium
```

File: tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl (hunk 1 above)
Comment (skanwal) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:77]: Noted. Think that'll be better - avoids confusion for using germline term in wts (tumor) pipeline.

Comment (skanwal) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:77]: I'll update it.

Comment (alexiswl) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:77]: This way you get to write some typescript!

File: tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl (hunk 2 above)
Comment (skanwal) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:31]: I actually initially used 

```
hints:
    ResourceRequirement:
        ilmn-tes:resources:tier: standard
        ilmn-tes:resources:type: fpga
        ilmn-tes:resources:size: medium
```

but this errored out in wfr.7fd3884c46dd4e95808bd25c6443ca53 with the hardware issue.

Comment (alexiswl) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:31]: Gah!

https://github.com/umccr-illumina/cwl-iap/issues/189#issuecomment-880913357

Needs to be 
```
hints:
    ResourceRequirement:
        ilmn-tes:resources/tier: standard
        ilmn-tes:resources/type: fpga
        ilmn-tes:resources/size: medium
```

Comment (alexiswl) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:31]: I'll update the rest of the tools we've just changed, my bad!

Comment (alexiswl) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:31]: Resolved in #200

**Comment by alexiswl** [comment]
Waiting for merge of #200

File: tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl (hunk 2 above)
Comment (skanwal) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl]: updated in https://github.com/umccr/cwl-ica/pull/199/commits/73eaae0c24c7c1bf067785ecc45e399434dab8f3

File: typescript-expressions/dragen-tools/4.0.3/tests/summary.txt [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-Lines        : 78.68% ( 96/122 )
+Functions    : 55% ( 11/20 )
+Lines        : 78.22% ( 97/124 )
 ================================================================================
 
 Test Suites: 2 passed, 2 total
 Tests:       26 passed, 26 total
 Snapshots:   0 total
-Time:        33.198 s
+Time:        4.617 s
```
Comment (alexiswl) [inline typescript-expressions/dragen-tools/4.0.3/tests/summary.txt:16]: Wow! This runs a lot faster on an M1!

**Review by alexiswl — APPROVED** [review]

**Comment by alexiswl** [comment]
Merged main into this branch and resolved conflicts and approved.

Note: latest release of CLI does take some time to complete the GH actions, something I'll work on when I'm back in the office next week

**Pull request merged**